lib/cerner/oauth1a/oauth_error.rb

```diff
@@ -27,19 +30,23 @@ class OAuthError < StandardError
       # oauth_parameters   - A String/Symbol or Array of Strings/Symbols containing the names of parameters that
       #                      are absent or rejected. This is should only be used when oauth_problem
       #                      is 'parameter_absent' or 'parameter_rejected' Optional.
+      # realm              - The protection realm associated with the error. Optional.
       def initialize(
         message,
         http_response_code = nil,
         oauth_problem = nil,
-        oauth_parameters = nil
+        oauth_parameters = nil,
+        realm = nil
       )
         @http_response_code = http_response_code
         @oauth_problem = oauth_problem
         @oauth_parameters = oauth_parameters ? Array(oauth_parameters) : nil
+        @realm = realm
 
         parts = []
         parts << message if message
         parts << "HTTP #{@http_response_code}" if @http_response_code
+        parts << "OAuth Realm #{@realm}" if @realm # TODO: Should this keep the order of the .initialize parameters?
```

Comment:
Should the parts match the initialize method parameter order or is putting realm up front better for matching the www-authenticate header structure?

Reply:
I'd like to just add to the end of the message, as that seems like the most passive change.

Reply:
Moved the realm to the end: https://github.com/cerner/cerner-oauth1a/pull/4/commits/45b1b8610b27881899bac4ead00cd30766cc307f